avocado/utils/memory.py

```diff
@@ -320,3 +320,87 @@ def get_thp_value(feature):
         return (re.search(r"\[(\w+)\]", value)).group(1)
     else:
         return value
+
+
+class MemInfo(object):
+    """
+    Representation of /proc/meminfo
+    """
+
+    class MemInfoObject(object):
+        """
+        Representation of one item from /proc/meminfo
+        """
+        def __init__(self, name, value):
+            self.name = name
+            self.value = int(value)
+            self.__units = {'bytes': 2**0,
+                            'kbytes': 2**10,
+                            'mbytes': 2**20,
+                            'gbytes': 2**30,
+                            'tbytes': 2**40}
```

Comment:
Well this is nice, but it requires 5x power to be evaluated on each MemInfoObject creation despite whether it'd ever beconverted or not. I'd prefer good old boring hardcoded methods which does not require any workarounds in __getattr_, no units dicts etc.